### PR TITLE
Force form type for some fields

### PR DESCRIPTION
**Pull request by ebuildy**
Fix exception ``Please define a type for field `firstname` in `Sonata\UserBundle\Admin\Entity\UserAdmin`` (I got this error since a few days, I didnt really dig, but this PR should fix it).

I am targeting this branch, because "I am using this branch and see a bug on it since a few days".

### CODE REVIEW

File: Admin/Model/UserAdmin.php [hunk 1]
```diff
@@ -209,8 +209,8 @@ protected function configureFormFields(FormMapper $formMapper)
                         'dp_max_date' => $now->format('c'),
                         'required' => false,
                     ])
-                    ->add('firstname', null, ['required' => false])
```
Comment (core23) [inline Admin/Model/UserAdmin.php]: Please remove the pre-sf2.8 hacks too:
https://github.com/ebuildy/SonataUserBundle/blob/195c5848cad92f0b67a73922b4cdb0fadb775bd7/Admin/Model/UserAdmin.php#L176-L194

**Comment by greg0ire** [comment]
Hey there!

It looks like you did not fill the pull request template properly. Maybe you
were confused, so let me walk you through this step by step.

# Explaining why you target this branch

> I am targeting this branch, because {reason}.

Here, you are supposed to replace `{reason}` with something like:

- "there is a BC-break" if you picked the master branch
- "there is no BC-break" if you picked the stable branch

# Searching and finding related issues

> Closes #{put_issue_number_here}

Please have a look at the list of issues and pull requests, and look for issues
that will be fixed, and pull requests that will become irrelevant should we
merge your PR. Each issue / pull request has a number shown right next to the
title. Kindly replace `{put_issue_number_here}` with whatever issue or PR is
relevant. Of course, you may specify several numbers.
On merge, [these issues and PRs will automatically get
closed](https://help.github.com/articles/closing-issues-using-keywords/).
If there is no issue or pull request that fits there, just remove the sentence
entirely.

# Filling the changelog

That one is really important to get right, because it will be used to craft a
release changelog such as [this
one](https://github.com/sonata-project/SonataAdminBundle/releases/tag/3.23.0).
The first thing you need to know about this is that if your pull request is
pedantic, meaning it does not fix a bug for the user, or adds a new feature, it
is not needed. This is typically the case for PRs that fix the build or add
some documentation.

3 important things:

1. Make sure to fill the changelog section with something directed at the end
user, that answers the question "What benefit do I get from upgrading?".
2. Make sure to remove empty sections.
3. Make sure to keep the markdown markup.

## Filling the TODO list

If you think your PR is ready, just remove it. If you have still some things
left to do, customize it.

## Filling the Subject

Last but not least, describe your PR as well as you can. Bear in mind that some
reviewers do not even use the library they are maintaining, so try to be
crystal clear.

**Comment by greg0ire** [comment]
> This branch cannot be rebased due to conflicts

Please rebase @ebuildy

**Comment by jordisala1991** [comment]
I dont understand why we need to add explicit types if there is a guesser that solves that. Im not against it but would be better to know if the guesser is broken or you are using an older version of sonatadoctrineorm or other sonata bundles

**Comment by OskarStark** [comment]
@ebuildy can you please show which sonata versions you are using?

I am 👍 with @jordisala1991, my code is working completely without this fix

**Pull request closed without merging**